components/ide/jetbrains/launcher/main.go

```diff
@@ -488,12 +490,25 @@ func resolveUserEnvs(launchCtx *LaunchContext) (userEnvs []string, err error) {
 	if err != nil {
 		return
 	}
-	envCmd := exec.Command(shell, []string{"-ilc", "/ide-desktop/jb-launcher env " + mark.String()}...)
+
+	self, err := os.Executable()
+	if err != nil {
+		return
+	}
+	envCmd := exec.Command(shell, []string{"-i", "-l", "-c", strings.Join([]string{self, "env", mark.String()}, " ")}...)
+	envCmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
 	envCmd.Stderr = os.Stderr
+	envCmd.WaitDelay = 3 * time.Second
```

Comment:
Do you think it would be enough? It basically depend on sourcing of bash scripts. Maybe better to be on safe side and put something like 10 secs?

Reply:
[From the docs](https://pkg.go.dev/os/exec) it starts counting once the process has exited :
> 	// If WaitDelay is non-zero, it bounds the time spent waiting on two sources
	// of unexpected delay in Wait: a child process that fails to exit after the
	// associated Context is canceled, and a child process that exits but leaves
	// its I/O pipes unclosed.
	//
	// The WaitDelay timer starts when either the associated Context is done or a
	// call to Wait observes that the child process has exited, whichever occurs
	// first. When the delay has elapsed, the command shuts down the child process
	// and/or its I/O pipes.
	//
	// If the child process has failed to exit — perhaps because it ignored or
	// failed to receive a shutdown signal from a Cancel function, or because no
	// Cancel function was set — then it will be terminated using os.Process.Kill.
	//
	// Then, if the I/O pipes communicating with the child process are still open,
	// those pipes are closed in order to unblock any goroutines currently blocked
	// on Read or Write calls.
	//
	// If pipes are closed due to WaitDelay, no Cancel call has occurred,
	// and the command has otherwise exited with a successful status, Wait and
	// similar methods will return ErrWaitDelay instead of nil.
	//
	// If WaitDelay is zero (the default), I/O pipes will be read until EOF,
	// which might not occur until orphaned subprocesses of the command have
	// also closed their descriptors for the pipes.